docs/feature_external_userspace.md

```diff
@@ -0,0 +1,69 @@
+# External Userspace: Store your code outside the QMK repo
```

Comment:
Would it make sense to also document how to use this with plain `make` invocations?

Reply:
I don't think so. 

While it can be used with `make`, absolutely... the overall goal is to remove the dependency on `make` as much as possible, and push everyone to `qmk`. 

Because of that, I don't think that adding documentation for make, well, makes any sense.

Reply:
Ok!